Chaincase.Common/Extensions.cs

```diff
@@ -16,21 +18,26 @@ public static class Extensions
     {
         public static IServiceCollection AddCommonServices(this IServiceCollection services)
         {
+	        services.AddOptions()
+		        .ConfigureOptions<Config>()
+		        .PostConfigure(new Action<Config>(config =>
+		        {
+			        config.MixUntilAnonymitySet = Config.GetNormalizeAnonSet(config);
+		        }))
+		        .ConfigureOptions<UiConfig>();
             services.AddSingleton<Global>();
-            services.AddSingleton<Config>();
-            services.AddSingleton<UiConfig>();
+            
+            // services.AddSingleton<UiConfig>();
             services.AddSingleton(x => {
-                var config = x.GetRequiredService<Config>();
-                // TODO FEEL ENOUGH PAIN FROM THIS ANTIPATTERN TO FIX IT
-                config.LoadOrCreateDefaultFile();
-                var network = config.Network;
+                var config = x.GetRequiredService<IOptions<Config>>();
+                var network = config.Value.Network;
                 var dataDir = x.GetRequiredService<IDataDirProvider>().Get();
                 var notificationManager = x.GetRequiredService<INotificationManager>();
                 return new ChaincaseWalletManager(network, new WalletDirectories(dataDir), notificationManager);
             });
             services.AddSingleton(x =>
             {
-                var network = x.GetRequiredService<Config>().Network;
+                var network = x.GetRequiredService<IOptions<Config>>().Value.Network;
```

Comment:
Network is a depencency on its own all over the place. Is there a way to set up a function in the DI container to just return Network from the loaded config? NetworkProvider?